### PR TITLE
Fixes #1697: Visibility of dates layout fixed in SavingAccountsTransactionFragment.kt

### LINKED ISSUE #1697: Unnecessary visibility of dates in Transactions Filter Dialog
**Summary:**
When date radiobutton is not selected even though start and end are disabled they are visible. This might not happen when buttons are disables instead layout should be hidden.

**Steps to reproduce:**
1 Open Savings Account
2 Open Transactions 
3 Open Filter Dialog

**Expected behaviour:**
Start and End date should be visible only when date radiobutton is clicked.

**Device and Android version:**
Any

**Screenshots:**

### PR DESCRIPTION
**Pull request by PatelVatsalB21**
Fixes #1697 
Now dates are visible only when date radiobutton is checked

https://user-images.githubusercontent.com/70195106/104495740-22de9a00-55fe-11eb-8793-f979084044a7.mp4

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt [hunk 1]
```diff
@@ -280,10 +279,12 @@ class SavingAccountsTransactionFragment : BaseFragment(), SavingAccountsTransact
                 isReady = false
                 radioGroupFilter?.clearCheck()
                 selectedRadioButtonId = -1
+                dateLayout?.visibility = View.GONE
             } else {
                 if (selectedRadioButtonId == -1) {
                     val btn = dialogView.findViewById<RadioButton>(R.id.rb_date)
                     btn.isChecked = true
+                    dateLayout?.visibility = View.VISIBLE
```
Comment (divyank00) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt]: No need for this line as line 314 will run

**Review by divyank00 — CHANGES_REQUESTED** [review]

**Comment by PatelVatsalB21** [comment]
@divyank00 thanks for suggestion. I have updated pr. Please have a look. Thank you:)

**Comment by divyank00** [comment]
LGTM

**Review by divyank00 — APPROVED** [review]

File: app/src/main/res/layout/layout_filter_dialog.xml [hunk 2]
```diff
@@ -79,6 +79,7 @@
                         android:text="@string/date" />
 
                     <LinearLayout
+                        android:id="@+id/start_end_date_layout"
```
Comment (miPlodder) [inline app/src/main/res/layout/layout_filter_dialog.xml]: use better standards of names, `ll_start_end_date`

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt [hunk 3]
```diff
@@ -265,10 +264,10 @@ class SavingAccountsTransactionFragment : BaseFragment(), SavingAccountsTransact
         val llcheckBoxPeriod = dialogView?.findViewById<LinearLayout>(R.id.ll_row_checkbox)
         val checkBoxPeriod: AppCompatCheckBox? = dialogView?.findViewById(R.id.cb_select)
         val radioGroupFilter = dialogView?.findViewById<RadioGroup>(R.id.rg_date_filter)
+        val dateLayout = dialogView?.findViewById<LinearLayout>(R.id.start_end_date_layout)
```
Comment (miPlodder) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt]: convert layout `id` in camelcase and use it as variable name

**Review by miPlodder — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt (hunk 3 above)
Comment (PatelVatsalB21) [reply app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt]: @miPlodder please review changes

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt [hunk 4]
```diff
@@ -265,10 +264,10 @@ class SavingAccountsTransactionFragment : BaseFragment(), SavingAccountsTransact
         val llcheckBoxPeriod = dialogView?.findViewById<LinearLayout>(R.id.ll_row_checkbox)
         val checkBoxPeriod: AppCompatCheckBox? = dialogView?.findViewById(R.id.cb_select)
         val radioGroupFilter = dialogView?.findViewById<RadioGroup>(R.id.rg_date_filter)
+        val llstartEndDate = dialogView?.findViewById<LinearLayout>(R.id.ll_start_end_date)
```
Comment (miPlodder) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt]: ll`S`, follow camelcase conv properly.

**Review by miPlodder — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt (hunk 4 above)
Comment (PatelVatsalB21) [reply app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt]: Ohh sorry I copied format from checkbox period. I am updating it.

Comment (PatelVatsalB21) [reply app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt]: @miPlodder review changes.

**Pull request closed without merging**